### DISCUSSION TITLE
Uploading summit slides

### DISCUSSION THREAD

**Discussion by olzama** [delph-in/docs#87] [Team Posts]
I added my first deck of slides to the docs repo as an experiment, to be able to link to it from the schedule. I created a folder called "slides" and a folder called "summits" and a folder called "2021" under "docs", and put it there. Does anyone have a better idea about a convention/path?

**Comment by goodmami** [comment]
I think the `slides` directory is not needed. Just `summits/2021/` would be good, in case there are uploads that aren't slides, and also to avoid unnecessary nesting, but otherwise that seems fine to me. The issue is that clicking the link opens a preview in GitHub's website and doesn't download the PDF. If you want more standard behavior, you have some options:

- Upload them as attachments on issues (no versioning, a bit of a hack)
- Put them in a directory or branch that is published with GitHub Pages, then use the github.io URL instead)

**Comment by olzama** [comment]
Yes actually publishing them with Github pages sounds like a good use of that resource.

**Comment by goodmami** [comment]
Oh there's another option:

- Replace `blob` with `raw` in the URL (e.g., https://github.com/delph-in/docs/raw/main/slides/summits/2021/DELPH_IN2021_wh.pdf instead of https://github.com/delph-in/docs/blob/main/slides/summits/2021/DELPH_IN2021_wh.pdf)

This method is probably the easiest. There are some benefits to publishing with GitHub Pages over this method but I don't think they are very relevant for us, and not worth the complexity.

**Comment by olzama** [comment]
That's definitely the easiest if we want an automatic download.
I now started to wonder though if just going with blob isn't better? For all practical purposes, it is the same as publishing the slides on a website (it is easy to look at them), and the "Download" button is right there.

**Comment by olzama** [comment]
There is a problem of course with storing files where everyone has push access, in that people may just call their slides "delphin2021" and overwrite other people's slides.

**Comment by goodmami** [comment]
> people may just call their slides "delphin2021" and overwrite other people's slides

Well, they could do more damage if they wanted to. Let's trust people to act responsibly/carefully? If there's a mistake, it's not hard to revert the change.

**Comment by olzama** [comment]
I think potentially, it can be "hard" in the sense that something like this will likely go unnoticed for years and then there will be frustrating broken links (suppose someone came across a link who is not the author and perhaps even not a member? They aren't gonna complain or try to fix it, they will just walk away).

Wasn't there a way to protect a particular branch?

**Comment by arademaker** [comment]
Late to this discussion. But we should have defined some convention for the names

firstname-day-shorttitle.pdf
